### PR TITLE
Use generally available JDK21 images

### LINKED ISSUE #346: Add GA JDK21 Alpine-based arm64 images
### What feature do you want to see added?

I would like to have a non-`preview` image based on Alpine Linux for arm64 architecture. This is now possible as base images of this type with JDK21 are now generally available from Eclipse Temurin.

### Upstream changes

The base image (Eclipse Temurin JDK21) was tracked in https://github.com/adoptium/containers/issues/158.
The main Jenkins image did the same change already, tracked in https://github.com/jenkinsci/docker/issues/1244.

### PR DESCRIPTION
**Pull request by marvinruder**
* Create separate `-preview` configurations for architectures in early access
* Fix early access URLs
* Switch to Ubuntu 22.04 LTS build images
* Fix `make` command in README.md

Resolves #346 

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by marvinruder** [comment]
Not sure if all my changes in the `updatecli` directory are as they should be, so please double-check there. 

I was able to build and test all images successfully on `arm64` architecture (including `-preview` images temporarily added to the configuration), but do not have access to other hardware, so a pipeline execution on your machines would be much appreciated.

**Comment by gounthar** [comment]
Thank you so much for your contribution, @marvinruder . 🤗 
I'll try a review tomorrow, but I guess others will start before I do. 😊

File: docker-bake.hcl [hunk 1]
```diff
@@ -5,6 +5,7 @@ group "linux" {
     "alpine_jdk21",
     "debian_jdk11",
     "debian_jdk21",
```
Comment (marvinruder) [inline docker-bake.hcl:7]: Not sure if `debian_jdk17` is missing here?

**Comment by marvinruder** [comment]
I also noticed that there is a `debian_jdk11` base image available for `linux/arm/v7`, and `debian_jdk17` base images available for `linux/s390x` and `linux/arm/v7`, but those combinations are not included in the [Docker bake file](https://github.com/jenkinsci/docker-ssh-agent/blob/master/docker-bake.hcl). Is this intentional, are there other incompatibilities I am unaware of?

**Comment by dduportal** [comment]
Hi @marvinruder ! Thanks for this PR!

> Not sure if all my changes in the `updatecli` directory are as they should be, so please double-check there.

No worries on the `updatecli` changes: the good practice is to focus on it on a subsequent PR so we'll check once this one is merged. Let's keep the current changes as they ensure we don't forget.

For the record, there are no execution errors in the CI (as per the GitHub Action check) for `updatecli`, which has 9 "pipelines" run, 5 are "success" (e.g. tracked dependencies are up-to-date) and 4 are "skipped" (new dependency detected but the condition are failing).

> I was able to build and test all images successfully on `arm64` architecture (including `-preview` images temporarily added to the configuration), but do not have access to other hardware, so a pipeline execution on your machines would be much appreciated.

Thanks for carefully checking! The pipeline checks are also all green which mean that the [(build + test)](https://github.com/jenkinsci/docker-ssh-agent/blob/0e5a76ea4d0ee75ab872cc000f485a1c1bf8b5cb/Jenkinsfile#L72-L77) on `amd64` are all good and the "multi-architecture" was also a success.

For the record, we do not use specifi hardware for cross-platform container image build: as per https://github.com/jenkinsci/docker-ssh-agent/blob/0e5a76ea4d0ee75ab872cc000f485a1c1bf8b5cb/Jenkinsfile#L78-L82 we use `qemu` inside Docker along with buildx capabilities. it's a 3 line (and portable if you have Docker CE) script. No action expected i'm only adding this here so you know it is possible on your machine :)


> I also noticed that there is a debian_jdk11 base image available for linux/arm/v7, and debian_jdk17 base images available for linux/s390x and linux/arm/v7, but those combinations are not included in the [Docker bake file](https://github.com/jenkinsci/docker-ssh-agent/blob/master/docker-bake.hcl). Is this intentional, are there other incompatibilities I am unaware of?

Good catch, but that should be a topic for another PR if you don't mind. Having biug PRs with too much things impairs the ability to review and integrate feature efficiently: Let's start with the JDK21. I propose we or you open an issue to point the debian jdk11 stuff: I believe we should check that the default image switches to JDK17 as default in a subsequent PR (which would be a breaking change to be delivered after the JDK21 GA).

**Review by dduportal — APPROVED** [review]
Nice work, the image changes looks good, let's roll!

**Pull request merged**